providers/unleash/pkg/provider.go

```diff
@@ -127,9 +127,21 @@ func (p *Provider) FloatEvaluation(ctx context.Context, flag string, defaultValu
 
 func (p *Provider) IntEvaluation(ctx context.Context, flag string, defaultValue int64, evalCtx of.FlattenedContext) of.IntResolutionDetail {
 	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	if strValue, ok := res.Value.(string); ok {
+		value, err := strconv.ParseInt(strValue, 10, 64)
+		if err == nil {
+			return of.IntResolutionDetail{
+				Value:                    value,
+				ProviderResolutionDetail: res.ProviderResolutionDetail,
+			}
+		}
+	}
 	return of.IntResolutionDetail{
-		Value:                    res.Value.(int64),
-		ProviderResolutionDetail: res.ProviderResolutionDetail,
+		Value: defaultValue,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			Reason:          of.ErrorReason,
+			ResolutionError: of.NewFlagNotFoundResolutionError(fmt.Sprintf("IntEvaluation type error for %s", flag)),
```

Comment:
Wouldn't a type mismatch error be a better fit?

Reply:
A type mismatch error would be more appropriate. I've updated FloatEvaluation to reflect that as well